HHS/OPRE-OPS
# OPS-2264: Editor Specific Alert for Change Request Approval/Rejection (Backend)

## What changed

* adds new `Notification` subclass `ChangeRequestNotification` that has the `ChangeRequest` for the notification
* uses this when creating notifications for change requests (in utils/change_requests.py) such as when the are approve or rejected
* includes the change request in the notification response
* adds `agreement_id` to the query params for notifications to find notifications related to an agreement
  * currently the only notifications that are related to agreements are  `ChangeRequestNotification` for `AgreementChangeRequests`
  * expect this to be use in the FE like `/notifications/?agreement_id={agreement_id}&oidc_id={user_oidc_id}&is_read=False` to find notifications for change requests that were sent to the current user for the agreement being shown that have not been read


## Issue

#2264 

## How to test

* run the tests, `test_notification.py/test_notifications_get_by_agreement_id` was added for this
* additionally you can run the app to create and approve change requests and view the network panel for the notifications to see CR notifications included with the CRs embedded


## Definition of Done Checklist
